src/engine.rs

```diff
@@ -311,8 +311,8 @@ impl Engine {
                         bytes,
                         wgpu::ImageDataLayout {
                             offset: 0,
-                            bytes_per_row: NonZeroU32::new(
-                                image_proxy.width * format.describe().block_size as u32,
+                            bytes_per_row: Some(
+                                image_proxy.width * format.block_size(None).unwrap(),
```

Comment:
```suggestion
                                image_proxy.width * format.block_size(None).expect("All supported ImageFormats have a valid block size"),
```
or words to that effect

Reply:
Done.